pycytominer/tests/test_cyto_utils/test_cells.py

```diff
@@ -525,6 +525,60 @@ def test_aggregate_subsampling_profile_compress_multiple_queries():
     pd.testing.assert_frame_equal(result, expected_result)
 
 
+def test_n_aggregation_memory_strata():
+
+    df_n1 = ap.aggregate_profiles(n_aggregation_memory_strata=1)
+    df_n2 = ap.aggregate_profiles(n_aggregation_memory_strata=2)
+    df_n3 = ap.aggregate_profiles(n_aggregation_memory_strata=3)
+    df_n_large = ap.aggregate_profiles(n_aggregation_memory_strata=1000)
+
+    pd.testing.assert_frame_equal(df_n1, df_n2)
+    pd.testing.assert_frame_equal(df_n1, df_n3)
+    pd.testing.assert_frame_equal(df_n1, df_n_large)
+
+
+def test_invalid_n_aggregation_memory_strata():
+    # expect an AssertionError when an invalid parameter value is specified
+    with pytest.raises(AssertionError):
+        ap.aggregate_profiles(n_aggregation_memory_strata=0)
+
+
+def test_sqlite_strata_conditions():
+
+    df = pd.DataFrame(
+        data={
+            "TableNumber": [[1], [2], [3], [4]],
+            "ImageNumber": [[1], [1, 2, 3], [1, 2], [1]],
+        }
+    )
+
+    n1_expected_output = [
+        "(TableNumber in (1) and ImageNumber in (1))",
+        "(TableNumber in (2) and ImageNumber in (1, 2, 3))",
+        "(TableNumber in (3) and ImageNumber in (1, 2))",
+        "(TableNumber in (4) and ImageNumber in (1))",
+    ]
+    out1 = _sqlite_strata_conditions(
+        df=df,
+        dtypes={"TableNumber": "integer", "ImageNumber": "integer"},
```

Comment:
will the `dtype` ever be a string?

Reply:
oh, "text" - I see below, gotcha!

Reply:
Yeah, it seems like the `dtype`s for SQL are `null`, `real`, `text`, `integer`, and `blob`.  But the TableNumber and ImageNumber values I think in practice are always of `integer` type, though I am not 100% sure